src/main/converter/opds.ts

```diff
@@ -176,6 +176,14 @@ export class OpdsFeedViewConverter {
             navigation = feed.Navigation.map((item) => {
                 return this.convertOpdsLinkToView(item);
             });
+
+            // concatenate all relative path to an absolute URL path
+            navigation = navigation.map((nav) => {
+                if (nav.url && !/[a-zA-Z]+\:\/\//.exec(nav.url)) {
+                    nav.url = `${url}${nav.url}`;
```

Comment:
This string concatenation will fail in cases where the "base" URL of the OPDS feed contains query parameters, e.g.:
* `url` = "https://domain.com/catalog?format=opds&lang=fr"
* `nav.url` = "category/fiction/latest"
* `${url}${nav.url}` => "https://domain.com/catalog?format=opds&lang=frcategory/fiction/latest"

Reply:
This string concatenation technique will fail in cases where the "relative" URL (i.e. `nav.url`) is in fact an absolute path reference, for example:
* `url` = `https://domain.com/catalog.opds`
* `nav.url` = `/catalog/best-sellers.atom`
* `${url}${nav.url}` => `https://domain.com/catalog.opds/catalog/best-sellers.atom`

Reply:
The current URL normalization / absolutization routine does not correctly handle "protocol-relative URLs", for example:
* `url` = `https://domain.com/catalog.opds`
* `nav.url` = `//domain.com/catalog/best-sellers.atom`
* `${url}${nav.url}` => `https://domain.com/catalog.opds//domain.com/catalog/best-sellers.atom`

(this is probably not a common case, but I do remember coming across an OPDS feed that contained such "relative" URLs)

Reply:
The proposed string concatenation technique will fail in the following case:

* `url` = `https://domain.com/catalog/main.opds`
* `nav.url` = `category/fiction/latest.atom` (same problem with `./category/fiction/latest.atom`)
* `${url}${nav.url}` => `https://domain.com/catalog/main.opdscategory/fiction/latest.atom`

(the resulting absolutized URL should be `https://domain.com/catalog/category/fiction/latest.atom`)

Reply:
The proposed string concatenation technique will also fail in the following case:

* `url` = `https://domain.com/main/catalog.opds`
* `nav.url` = `../category/fiction/latest.atom`
* `${url}${nav.url}` => `https://domain.com/main/catalog.opds../category/fiction/latest.atom`

(the resulting absolutized URL should be `https://domain.com/category/fiction/latest.atom`)

Reply:
thanks daniel
I used url.resolve instead string concatenation

fixed https://github.com/readium/readium-desktop/pull/639/commits/8072856b3b320d8b3b3de70c1696580ee5eabaa2

Reply:
Passes the tests okay.

`node -e "const url = require('url'); const a = 'https://domain.com/catalog?format=opds&lang=fr'; const b = 'category/fiction/latest?param=test'; console.log(url.resolve(a, b));"`
=>
`https://domain.com/category/fiction/latest?param=test`

`node -e "const url = require('url'); const a = 'https://domain.com/catalog.opds'; const b = '/catalog/best-sellers.atom'; console.log(url.resolve(a, b));"`
=>
`https://domain.com/catalog/best-sellers.atom`

`node -e "const url = require('url'); const a = 'https://domain.com/catalog.opds'; const b = '//domain.com/catalog/best-sellers.atom'; console.log(url.resolve(a, b));"`
=>
`https://domain.com/catalog/best-sellers.atom`

`node -e "const url = require('url'); const a = 'https://domain.com/catalog/main.opds'; const b = 'category/fiction/latest.atom'; console.log(url.resolve(a, b));"`
=>
`https://domain.com/catalog/category/fiction/latest.atom`

`node -e "const url = require('url'); const a = 'https://domain.com/main/catalog.opds'; const b = '../category/fiction/latest.atom'; console.log(url.resolve(a, b));"`
=>
`https://domain.com/category/fiction/latest.atom`